### ISSUE TITLE
Parsing huge queries extremely slow some times?

### ISSUE THREAD

**Issue by simen** [sanity-io/gradient#617]
Querying an empty database this executes in millis:

```
*[slug.current == 'more-og-romsdal'] 
```
While this takes >1000ms

```
*[slug.current == 'more-og-romsdal'] | {
      ...,
      metadata{..., "ogImage": ogImage{"asset": asset->url}},
      "document": document -> { title, description,"file": file{"asset": asset->url}},
      program{ content[]{..., "file": file{"asset": asset->url}}},
      "redirect": redirect->{slug},
      "related_issues": related_issues[]->{title, slug, "redirect": redirect->{title, slug}},
      "thumbnail": thumbnail{"asset": asset->{url}},
      "headerImage": image{..., "asset": asset->{url}},
      "chapterHeaderImage": headerImage{image{..., "asset": asset->{url}}},
      "groups": groups[]{title, "members": members[]{role, "person": person->{name, phone, email, facebook, twitter, instagram, "profile_image": profile_image{crop, hotspot, "asset": asset->{url, metadata}}}}},
      "county": county->{..., title, slug},
      contact{..., "leader": leader->{..., "profile_image": profile_image{crop, hotspot, "asset": asset->{url, metadata}}}},
      "blocks": blocks{
        "content": content[]{
          ...,
          "file": file{"asset": asset -> url},
          "document": document -> {
            title,
            description,
            "file": file{"asset": asset->url}
          },
          "image": image{..., "asset": asset->{url, metadata}},
          "person": person->{
            ...,
            "profile_image": profile_image {
              crop,
              hotspot,
              "asset": asset->{url, metadata}
            }
          },
          "members": members[] {
            role,
            "person": person->{
              ...,
              "profile_image": profile_image{crop, hotspot, "asset": asset->{url, metadata}}
            }
          }
        }
      }
    }
```

**Comment by erikgrinaker** [comment]
A profile of this query shows:

```
Showing top 10 nodes out of 62
      flat  flat%   sum%        cum   cum%
     7.83s 32.20% 32.20%     22.39s 92.06%  github.com/sanity-io/gradient/pkg/search/plans.Walk /Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/search/plans/utils.go
     4.48s 18.42% 50.62%     22.39s 92.06%  github.com/sanity-io/gradient/pkg/search/plans.isLimited.func1 /Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/search/plans/utils.go
     3.82s 15.71% 66.32%     22.39s 92.06%  github.com/sanity-io/gradient/pkg/search/plans.Contains.func1 /Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/search/plans/utils.go
     2.98s 12.25% 78.58%      2.98s 12.25%  runtime.duffcopy /usr/local/Cellar/go/1.9.3/libexec/src/runtime/duff_amd64.s
     1.41s  5.80% 84.38%     22.39s 92.06%  github.com/sanity-io/gradient/pkg/search/plans.walkNodes /Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/search/plans/utils.go
     0.91s  3.74% 88.12%      0.91s  3.74%  runtime.usleep /usr/local/Cellar/go/1.9.3/libexec/src/runtime/sys_darwin_amd64.s
     0.70s  2.88% 91.00%     22.39s 92.06%  github.com/sanity-io/gradient/pkg/search/plans.Contains /Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/search/plans/utils.go
     0.68s  2.80% 93.79%     22.39s 92.06%  github.com/sanity-io/gradient/pkg/search/plans.isLimited /Users/erik/Projects/go/src/github.com/sanity-io/gradient/pkg/search/plans/utils.go
     0.49s  2.01% 95.81%      0.49s  2.01%  runtime.duffzero /usr/local/Cellar/go/1.9.3/libexec/src/runtime/duff_amd64.s
     0.19s  0.78% 96.59%      0.19s  0.78%  time.now /usr/local/Cellar/go/1.9.3/libexec/src/runtime/sys_darwin_amd64.s
```

**Comment by erikgrinaker** [comment]
This is ultimately caused by the `plans.AddDefaultLimit()` call in `planner.Build()`. Commenting it out reduces the query time from 1205ms to 75ms.

**Comment by erikgrinaker** [comment]
Fixed in 8dd1d65bf0620087e6c67fedac298bbf42b10b85
